scripts/scil_tractogram_filter_collisions.py

```diff
@@ -0,0 +1,255 @@
+#! /usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Given an input tractogram and a text file containing a diameter for each
+streamline, filters all intersecting streamlines and saves the resulting
+tractogram and diameters.
+
+IMPORTANT: The input tractogram needs to have been resampled to segments of
+at most 0.2mm. Otherwise performance will drop significantly.
+(see scil_tractogram_resample_nb_points.py)
+
+The filtering is deterministic and follows this approach:
+    - Pick next streamline
+    - Iterate over its segments
+        - If current segment collides with any other streamline segment given
+          their diameters
+            - Current streamline is deemed invalid and is filtered out
+            - Other streamline is left untouched
+    - Repeat
```

Comment:
Imagine we have two bundles intersecting eachother and generating collisions. Let's say that streamlines are saved in this order [BundleA_Streamline1, ..., BundleA_StreamlineN, BundleB_Streamline1, ..., BundleB_StreamlineM], everytime there's a collisions between the two bundles, it's "BundleA's fault" because it comes first. And Bundle A would get completely filtered out and Bundle B left untouched. It happens with the Fibercup, only one or two bundles were almost gone and the rest of the tractogram untouched.

Shuffling solves this perfectly :)